treeherder/model/management/commands/cycle_data.py

```diff
@@ -320,6 +321,83 @@ def __lookup_new_signature(self):
             raise LookupError('Exhausted all signatures originating from try repository.')
 
 
+class IrrelevantDataRemoval(RemovalStrategy):
+    """
+    Removes `performance_datum` rows
+    that originate from repositories other than
+    `autoland`, `mozilla-central`, `fenix`, `reference-browser` and
+    `mozilla-beta` that are more than 6 months old.
+    """
+
+    repositories = None
+    repositories_ids = []
+    repository_names = [
+        'autoland',
+        'mozilla-central',
+        'mozilla-beta',
+        'fenix',
+        'reference-browser',
+    ]
+
+    def __init__(self, chunk_size: int):
+        self._cycle_interval = timedelta(days=(6 * 30))
+        self._chunk_size = chunk_size
+        self._max_timestamp = datetime.now() - self._cycle_interval
+        self._manager = PerformanceDatum.objects
+
+    @property
+    def relevant_repositories(self):
+        if self.repositories is None:
+            self.repositories = Repository.objects.filter(name__in=self.repository_names).values(
+                'id'
+            )
+            if self.repositories.count() != len(self.repository_names):
+                logger.warning("Failed to find all relevant repositories in the database")
+        return self.repositories
+
+    @property
+    def name(self) -> str:
+        return 'irrelevant data removal strategy'
+
+    def remove(self, using: CursorWrapper):
+        """
+        @type using: database connection cursor
+        """
+        chunk_size = self._find_ideal_chunk_size()
+
+        for repository in self.relevant_repositories:
+            self.repositories_ids.append(repository['id'])
+
+        using.execute(
+            '''
+                DELETE FROM `performance_datum`
+                WHERE repository_id NOT IN %s AND push_timestamp <= %s
+                LIMIT %s
+            ''',
+            [
+                tuple(self.repositories_ids),
+                self._max_timestamp,
+                chunk_size,
+            ],
+        )
+
+    def _find_ideal_chunk_size(self) -> int:
+        max_id_of_non_expired_row = (
+            self._manager.filter(push_timestamp__gt=self._max_timestamp)
+            .exclude(repository_id__in=self.relevant_repositories)
+            .order_by('-id')[0]
+            .id
+        )
+        older_ids = (
```

Comment:
This name isn't technically correct. It stores a list of older perf data, rather than a list of ids.